### PR TITLE
Added bootaftercreate option

### PR DESCRIPTION
**Pull request by ravager-dk**
Added new bootaftercreate optional argument

Fixed cap

Do not wait for BootAfterCreate false

Added bootaftercreate documentation

Suppress bootaftercreate changes after creation

Added note for wait_on_ip with bootaftercreate

### CODE REVIEW

**Comment by ravager-dk** [comment]
Somewhat related to #263

File: xoa/resource_xenorchestra_vm.go [hunk 1]
```diff
@@ -244,6 +244,13 @@ $ xo-cli xo.getAllObjects filter='json:{"id": "cf7b5d7d-3cd5-6b7c-5025-5c935c8cd
 			Description: "Whether terraform should wait until IP addresses are present on the VM's network interfaces before considering it created. This only works if guest-tools are installed in the VM. Defaults to false.",
 			Optional:    true,
 		},
+		"bootaftercreate": &schema.Schema{
+			Type:             schema.TypeBool,
+			Default:          true,
+			Description:      "Whether Terraform should boot the VM after creation. Defaults to true.",
```
Comment (ddelnano) [inline xoa/resource_xenorchestra_vm.go:250]: We are currently partially migrated to tfplugindocs (see #256 for details), but that means that the `Description` string here should match what is in the `docs/resources/vm.md` file. That will make the eventual migration easier.

File: client/vm.go [hunk 2]
```diff
@@ -549,8 +550,26 @@ func (c *Client) waitForVmState(id string, stateConf StateChangeConf) error {
...
+			return vm, vm.PowerState, nil
+		}
+		stateConf := &StateChangeConf{
+			Pending: []string{"Running"},
+			Refresh: refreshFn,
+			Target:  []string{"Halted", "Stopped"},
+			Timeout: timeout,
+		}
+		_, err := stateConf.WaitForState()
+		return err
```
Comment (ddelnano) [inline client/vm.go:571]: This function is already a mess (which I'm guilty of :upside_down_face:), so I think in order to support managing `power_state` we need to leave this better than it currently is. My proposal for that was what I did in my partially implemented `power_state` change [here](https://github.com/terra-farm/terraform-provider-xenorchestra/compare/master...update-power_state-to-be-non-computed-vm-resource-argument#diff-27b07f0cb784dad37043e3062a4d86ed7ffc4029ac2e568eec2888b7db39b619R559-R570). 

Also from a quick glance, `Stopped` doesn't seem to be a real state ([source](https://github.com/vatesfr/xen-orchestra/blob/7c009b0fc0be09cb86045358b48cdf5a51d9f9b3/%40xen-orchestra/lite/src/libs/xen-api/xen-api.enums.ts#L67-L72)). So references to that should ideally be cleaned up. That is dependent on checking the XAPI source to verify that those are a complete set of power states.

**Review by ddelnano — COMMENTED** [review]
I still think we should proceed with the full fledged `power_state` support I mentioned here (https://github.com/terra-farm/terraform-provider-xenorchestra/issues/263#issuecomment-1776591586), but I still wanted to provide feedback and appreciate your effort on this!

**Comment by ddelnano** [comment]
@ravager-dk I'm closing this since I believe #281 and #278 accomplished what you were hoping for. Please let me know if that isn't the case or you have more comments to discuss.

**Pull request closed without merging**